Update linux-raw-sys to 0.3.4

heiher opened the pull request:
To support LoongArch.

Changes:
```
Updating errno v0.2.8 -> v0.3.1
Updating is-terminal v0.4.4 -> v0.4.7
Updating linux-raw-sys v0.1.4 -> v0.3.4
Updating rustix v0.36.5 -> v0.37.7
Updating terminal_size v0.2.3 -> v0.2.6
  Adding windows-sys v0.48.0
```

The changes are generated by:

```bash
cargo update -p is-terminal -p terminal_size
```

Mark-Simulacrum commented on Cargo.lock:
```diff
@@ -5293,13 +5293,13 @@ version = "0.42.0"
+ "windows_x86_64_msvc 0.42.2",
 ]
 
 [[package]]
```
Should we / can we also bump this while we're at it? I'm not sure if that's a no-op change but it would be nice to avoid duplicating all these crates in our dependency tree.

heiher replied:
@Mark-Simulacrum Could you clarify if the objective is to minimize changes to unrelated dependencies and focus solely on upgrading linux-raw-sys? Would you be able to provide me with some guidance on how to proceed? Thank you. :heart:

heiher replied:
Were these changes generated by `cargo update -p is-terminal -p terminal_size` without any manual modifications. Should we trust the results generated by `cargo`? @Mark-Simulacrum

rust-log-analyzer commented:
The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/4751638234/jobs/8441008924) [(plain)](https://github.com/rust-lang/rust/commit/ccc939fc4e4edebfa9d880a12522bfccc45c8a4f/checks/12887165304/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
######################################################################## 100.0%
extracting /checkout/obj/build/cache/llvm-7fde08365c4c3cfc9a9f8a6a02693b789adbe6da-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
Building tool expand-yaml-anchors (stage0)
    Updating crates.io index
error: the lock file /checkout/Cargo.lock needs to be updated but --locked was passed to prevent this
If you want to try to generate the lock file without accessing the network, remove the --locked flag and use --offline instead.
```

</details>

heiher committed:
Update dependencies

To support LoongArch.

Changes:
  Updating errno v0.2.8 -> v0.3.1
  Updating is-terminal v0.4.4 -> v0.4.7
  Updating linux-raw-sys v0.1.4 -> v0.3.4
  Updating rustix v0.36.5 -> v0.37.7
  Updating terminal_size v0.2.3 -> v0.2.6
    Adding windows-sys v0.48.0

The changes are generated by:

  cargo update -p is-terminal -p terminal_size

bors commented:
:umbrella: The latest upstream changes (presumably #110186) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

heiher commented:
@Mark-Simulacrum After rebasing to https://github.com/rust-lang/rust/pull/110186, the patch appears to be more concise. Could you please review it again? Thank you.

bors commented:
:hourglass: Testing commit 7aa257f0ab72bae333e71fd0d88136ea6d132699 with merge 666fee2a5fb3a4ec75953b9e6622662ae10b9ae5...
<!-- homu: {"type":"BuildStarted","head_sha":"7aa257f0ab72bae333e71fd0d88136ea6d132699","merge_sha":"666fee2a5fb3a4ec75953b9e6622662ae10b9ae5"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4799602052/jobs/8542158146)
Approved by: Mark-Simulacrum
Pushing 666fee2a5fb3a4ec75953b9e6622662ae10b9ae5 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"Mark-Simulacrum","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/4799602052/jobs/8542158146"},"merge_sha":"666fee2a5fb3a4ec75953b9e6622662ae10b9ae5"} -->

rust-timer commented:
Finished benchmarking commit (666fee2a5fb3a4ec75953b9e6622662ae10b9ae5): [comparison URL](https://perf.rust-lang.org/compare.html?start=a7aa20517c80161a2ffe7c0c25fc2e0140c43c90&end=666fee2a5fb3a4ec75953b9e6622662ae10b9ae5&stat=instructions:u).

### Overall result: no relevant changes - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=a7aa20517c80161a2ffe7c0c25fc2e0140c43c90&end=666fee2a5fb3a4ec75953b9e6622662ae10b9ae5&stat=instructions:u)
This benchmark run did not return any relevant results for this metric.

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=a7aa20517c80161a2ffe7c0c25fc2e0140c43c90&end=666fee2a5fb3a4ec75953b9e6622662ae10b9ae5&stat=max-rss)
<details>
<summary>Results</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
| Improvements ✅ <br /> (primary)   | -     | -              | 0     |
| Improvements ✅ <br /> (secondary) | -0.7% | [-0.7%, -0.7%] | 1     |
| All ❌✅ (primary)                 | -     | -              | 0     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=a7aa20517c80161a2ffe7c0c25fc2e0140c43c90&end=666fee2a5fb3a4ec75953b9e6622662ae10b9ae5&stat=cycles:u)
This benchmark run did not return any relevant results for this metric.
